services/ec2.py

```diff
@@ -2,58 +2,191 @@
 import boto3
 
 
-# client = boto3.client("")
+client = boto3.client("ec2")
+autoscaling_client = boto3.client("autoscaling")
+ssm_client = boto3.client("ssm")
 
 
 def autoscaling_launch_template():
+    compliant_resources = []
+    non_compliant_resources = []
+    asgs = autoscaling_client.describe_auto_scaling_groups()["AutoScalingGroups"]
+
+    for asg in asgs:
+        if "LaunchTemplate" in asg["MixedInstancesPolicy"]:
+            compliant_resources.append(asg["AutoScalingGroupARN"])
+        else:
+            non_compliant_resources.append(asg["AutoScalingGroupARN"])
```

Comment:
`asg["MixedInstancesPolicy"]["LaunchTemplate"]`뿐만 아니라 `asg["LaunchTemplate"]`에 정의되어 있는 경우도 있습니다.
반대로, launch configuration을 사용했는 지 검사할 수도 있습니다.
```suggestion
        if "LaunchConfigurationName" in asg:
            non_compliant_resources.append(asg["AutoScalingGroupARN"])
        else:
            compliant_resources.append(asg["AutoScalingGroupARN"])
```